src/ad-tracking/cta-tracker.ts

```diff
@@ -0,0 +1,60 @@
+import { communicationService, globalAction, ofType } from '@ad-engine/communication';
+import { AdSlot, FuncPipeline, FuncPipelineStep, slotService } from '@ad-engine/core';
+import { props } from 'ts-action';
+
+type AdStatus =
```

Comment:
It looks clean TBH :) Also, it follows our good practice of [type over enum](https://fandom.atlassian.net/wiki/spaces/ADEN/pages/506364058/TypeScript+Style+Guide#TypeScriptStyleGuide-TypeoverEnum) 👍

The only part I'm not sure of... is the first pipe necessary, doesn't it cause any side effect? I'd without the confusing pipe at the beginning ;)
```suggestion
type AdStatus = 'cm_register_impression'
	| 'cm_register_clicked'
	| 'cm_newsletter_impression'
	| 'cm_newsletter_clicked'
	| 'cm_fanlab_impression'
	| 'cm_fanlab_clicked';
```